webapp/sources/rudder/rudder-rest/src/test/scala/com/normation/rudder/rest/RestTestSetUp.scala

```diff
@@ -1053,6 +1034,40 @@ class RestTest(liftRules: LiftRules) {
     }
   }
 
+  /*
+   * Correctly build and scope mutable things to use the request in a safe
+   * way in the context of LiftRules.
+   */
+  def doReqZIO[T](mockReq: MockHttpServletRequest)(tests: Req => TestResult): TestResult = {
+    LiftRulesMocker.devTestLiftRulesInstance.doWith(liftRules) {
+      MockWeb.useLiftRules.doWith(true) {
+        MockWeb.testReq(mockReq)(tests)
+      }
+    }
+  }
+
+  /**
+   * Execute the request and get the response.
+   * The request must be a stateless one, else a failure
+   * will follow.
+   */
+  def execRequestResponseZIO[T](mockReq: MockHttpServletRequest)(tests: Box[LiftResponse] => TestResult): TestResult = {
+    doReqZIO(mockReq) { req =>
+      // the test logic is taken from LiftServlet#doServices.
+      // perhaps we should call directly that methods, but it need
+      // much more set-up, and I don't know for sure *how* to set-up things.
+      NamedPF
+        .applyBox(req, LiftRules.statelessDispatch.toList)
+        .map(_.apply() match {
+          case Full(a) => Full(LiftRules.convertResponse((a, Nil, S.responseCookies, req)))
+          case r       => r
+        }) match {
+        case Full(x) => tests(x)
+        case eb: EmptyBox => tests(eb)
+      }
+    }
+  }
+
```

Comment:
these two methods are just the copy/paste of the corresponding methods for specs, but for zio-test.